python/ipyleaflet_core/ipyleaflet/leaflet.py

```diff
@@ -2215,6 +2192,135 @@ def clear_markers(self):
+
+    def _handle_leaflet_event(self, _, content, buffers):
+        if content.get("event", "").startswith("draw"):
+            event, action = content.get("event").split(":")
+            self.last_draw = content.get("geo_json")
+            self.last_action = action
+            self._draw_callbacks(self, action=action, geo_json=self.last_draw)
+
+
+class GeomanDrawControl(DrawControlBase):
+    """GeomanDrawControl class.
+
+    Alternative drawing tools for drawing on the map provided by Leaflet-Geoman.
+    """
+
+    _view_name = Unicode("LeafletGeomanDrawControlView").tag(sync=True)
+    _model_name = Unicode("LeafletGeomanDrawControlModel").tag(sync=True)
+
+    # Current mode & shape
+    # valid values are: 'draw', 'edit', 'drag', 'remove', 'cut', 'rotate'
+    # for drawing, the tool can be added after ':' e.g. 'draw:Marker'
+    current_mode = Any(allow_none=True, default_value=None).tag(sync=True)
+
+    # Hides toolbar
+    hide_controls = Bool(False).tag(sync=True)
+
+    circlemarker = Dict({ 'shapeOptions': {} }).tag(sync=True)
+
+    # Disabled by default
+    text = Dict().tag(sync=True)
+
+    # Tools
+    # See https://www.geoman.io/docs/modes
+    drag = Bool(True).tag(sync=True)
+    cut = Bool(True).tag(sync=True)
+    rotate = Bool(True).tag(sync=True)
+
+    def __init__(self, **kwargs):
+        super(GeomanDrawControl, self).__init__(**kwargs)
+        self.on_msg(self._handle_leaflet_event)
+
+    def _handle_leaflet_event(self, _, content, buffers):
+        if content.get('event', '').startswith('pm:'):
+            action = content.get('event').split(':')[1]
+            geo_json = content.get('geo_json')
+            if action == "vertexadded":
+                self._draw_callbacks(self, action=action, geo_json=geo_json)
+                return
+            # Some actions return only new feature, while others return all features
+            # in the layer
+            if not isinstance(geo_json, list):
+                geo_json = [geo_json]
+            self._draw_callbacks(self, action=action, geo_json=geo_json)
+
+    def on_draw(self, callback, remove=False):
+        """Add a draw event listener.
+
+        Parameters
+        ----------
+        callback : callable
+            Callback function that will be called on draw event.
+        remove: boolean
+            Whether to remove this callback or not. Defaults to False.
+        """
+        self._draw_callbacks.register_callback(callback, remove=remove)
+
+    def clear_text(self):
+        """Clear all text."""
+        self.send({'msg': 'clear_text'})
+
+
+class DrawControlCompatibility(DrawControlBase):
+    """DrawControl class.
+
+    Python side compatibility layer for old DrawControls, using the new Geoman front-end but old Python API.
+    """
+    
+    _view_name = Unicode("LeafletGeomanDrawControlView").tag(sync=True)
+    _model_name = Unicode("LeafletGeomanDrawControlModel").tag(sync=True)
+
+    circlemarker = Dict().tag(sync=True)
```

Comment:
In the default value should match what it was originally, this can go into the base class (if not, ignore my other comment on the same property in the Geoman class)

Reply:
Indeed this should match the original case. However I've put them separately for the different classes, since for the geoman implementation the default options should be `'pathOptions': {}`, while for the other two they should be `'shapeOptions': {}` (although for the geoman implementations either will work).